source/develop/release-management/features/virt/vm-lifecycle-with-kubevirt.md

```diff
@@ -0,0 +1,130 @@
+## Milestone 1: Basic integration
+
+### Goal
+1. Add Kubevirt into oVirt.  
+2. Kubevirt sends the run-request to the relevant node.  
+
+### Design
+We aim to have Kubevirt as a configuration of a cluster so eventually we would be able to have a Data-center with cluster with lower compatibility version that works as it used to work before and cluster with the up-to-date compatibility version that works with Kubevirt. That way, we enable the user to upgrade existing system without having to stop the VMs.  
+
+So a cluster would have a property called "type". Initially the type will have two values: legacy, kubevirt. If 'kubvirt' is chosen, the user will need to provide the URL and authentication details of Kubevirt.  
+
+Note that we may use the same mechanism used in external provider to hold the authentication info. Another alternative is to use a full blown external provider for this, as we did for the integration with virt-v2v (where external provider is used only to contain pre-defined configuration of the remote system). Anyway, we do not have to decide this at this stage.  
+
+Then, the mentioned run-VM flow will be modified. The only change would be that in Kubvirt-cluster, CreateBrokerVDSCommand will generate a bit different configuration of the VM and will send it to Kubevirt (using the Kubernetes API) rather than VDSM. Note that the scheduling is still done in oVirt.  
+
+The represenation of the VM will be as follow (in JSON):  
+1. Domxml (Kubevirt passes it as-is to the node at this stage, later on Kubervirt will replace placeholders within the XML).  
+2. The scheduled host (Kubvirt uses this as the destination node).
+
+### Benefits
+1. First notion of Kubernetes in oVirt + start getting feedback on running real-world virtual machines with Kubevirt.    
+2. We bypass the 'create' verb in the engine<->vdsm API.  
+3. We do not depend on having a full blown scheduler in Kubervirt.
+
+### Limitations & Challenges
+1. At this stage the engine does not do the operations that it used to do after the detination host was chosen and before sending the request to VDSM.  
+2. The engine cannot lock the monitoring until VDSM acknowlege that the request arrives. So we need to set a timeout so if the VM is still down afterwards, we'll conclude that the run operation failed.
+3. oVirt still schedules the nodes.
+4. oVirt still monitors the nodes as it used before.  
+5. Most of the VM related operations are still done directly with VDSM (rather than through Kubervirt).
+
+## Milestone 2: Scheduling
+
+### Goal
+Basic VM schedluing in Kubevirt.
+
+### Design
+1. Extend the VM representation with cluster-related data, e.g., host-pinning. This will be represented as properties in JSON. VM-related data needs to be retrieved from the VM configuration (see VM Represenation section).  
+2. Adjust VMs monitoring as described above. That is, to add a timeout for run-vm so the monitoring will end the run-vm operation if the VM is down after some time.  
```

Comment:
I would not be eager to do that right away. Since the critical part is libvirt stats I'd rather stay as close to it as possible.
Current ovirt's collectd(added for metrics) is not containerized ATM so I'd just follow that (ansible deployment). 
We can revise that later on, it's not important